shell/lib/db.js

```diff
@@ -321,6 +331,18 @@ isAdmin = function() {
   }
 }
 
+isAdminById = function(id) {
+  check(id, String);
```

Comment:
Maybe just return false if id is undefined? Otherwise I think all the places where you used this will throw if used by a non-logged-in user.